source/logrepl/snapshot.go

```diff
@@ -133,7 +141,15 @@ func (s *SnapshotIterator) Teardown(ctx context.Context) error {
 	if commitErr := s.tx.Commit(ctx); commitErr != nil {
 		sdk.Logger(ctx).Err(commitErr).Msg("teardown commit failed")
 	}
-	return s.rows.Err()
+	if rowsErr := s.rows.Err(); rowsErr != nil {
+		sdk.Logger(ctx).Err(rowsErr).Msg("rows returned an error")
+	}
+
```

Comment:
Question: Should we return a multierror here or is logging enough?

Reply:
If we only log those first two errors then they are not propagated to the caller in case everything else succeeds, that's not ideal. We could use a multierror lib, although what if we simply return the first error that occurs and log any error that happens after that? Something like this:
```
...
var err error
if commitErr := s.tx.Commit(ctx); commitErr != nil {
	err = logOrReturnError(ctx, err, commitErr, "teardown commit failed")
}
...

func logOrReturnError(ctx context.Context, oldErr, newErr error, msg string) error {
	if oldErr == nil {
		return fmt.Errorf(msg+": %w", newErr)
	}
	sdk.Logger(ctx).Err(newErr).Msg(msg)
	return oldErr
}
```

Playground example: https://go.dev/play/p/lua_Yjb4avR